build-logic/src/main/kotlin/org.sonarsource.iac.java-conventions.gradle.kts

```
@@ -0,0 +1,34 @@
+plugins {
+    `java-library`
+}
+
+java {
+    withSourcesJar()
+    withJavadocJar()
+    sourceCompatibility = JavaVersion.VERSION_11
+    targetCompatibility = JavaVersion.VERSION_11
+}
+
+group = "org.sonarsource.iac"
+version = "1.24-SNAPSHOT"
+
+tasks.withType<JavaCompile> {
+    options.encoding = "UTF-8"
+}
+
+tasks.withType<Javadoc> {
+    options.encoding = "UTF-8"
+    options {
+        (this as CoreJavadocOptions).addStringOption("Xdoclint:none", "-quiet")
```

Shouldn't it be `-Xdoclint:none`?

---

To be honest, I took this file from sonar-text :sweat_smile:
From the documentation, the first parameter is the option name and does not need the `-`, the method is taking care to add it.
The second parameter is supposed to be the value, but here it looks like more that it's another option.
I would change it to this in the end:
```suggestion
        (this as CoreJavadocOptions).addStringOption("Xdoclint:none")
        this.addStringOption("quiet")
```
WDYT?

---

Hm, apparently, Gradle prepends `-` inside `JavadocOptionFileWriterContext`. Yes, I think your variant with two calls seems more correct.

---

This is apparently not working.
I found an explanation why [here](https://discuss.gradle.org/t/how-to-send-x-option-to-javadoc/23384/4).
So I revert back to the first version, this is one of the only tricks to make it work :neutral_face: